FIAT/brezzi_douglas_marini_cube.py

```diff
@@ -0,0 +1,229 @@
+
+        for j in sorted(flat_topology[1]):
+            entity_ids[1][j] = list(range(cur, cur + degree + 1))
+            cur = cur + degree + 1
+
+        entity_ids[2][0] = list(range(cur, cur + 2*triangular_number(degree - 1)))
+        cur += 2*triangular_number(degree - 1)
+
+        formdegree = 1
+
+        entity_closure_ids = make_entity_closure_ids(flat_el, entity_ids)
+
+        super(BrezziDouglasMariniCube, self).__init__(ref_el=ref_el, dual=None, order=degree, formdegree=formdegree,
+                                                      mapping=mapping)
+
+        topology = ref_el.get_topology()
+        unflattening_map = compute_unflattening_map(topology)
+        unflattened_entity_ids = {}
+        unflattened_entity_closure_ids = {}
+
+        for dim, entities in sorted(topology.items()):
+            unflattened_entity_ids[dim] = {}
+            unflattened_entity_closure_ids[dim] = {}
+        for dim, entities in sorted(flat_topology.items()):
+            for entity in entities:
+                unflat_dim, unflat_entity = unflattening_map[(dim, entity)]
+                unflattened_entity_ids[unflat_dim][unflat_entity] = entity_ids[dim][entity]
+                unflattened_entity_closure_ids[unflat_dim][unflat_entity] = entity_closure_ids[dim][entity]
+        self.entity_ids = unflattened_entity_ids
+        self.entity_closure_ids = unflattened_entity_closure_ids
+        self._degree = degree
+        self.flat_el = flat_el
+
+    def degree(self):
+        return self._degree
+
+    def get_nodal_basis(self):
+        raise NotImplementedError("get_nodal_basis not implemented for bdmce")
+
+    def get_dual_set(self):
+        raise NotImplementedError("get_dual_set is not implemented for bdmce")
+
+    def get_coeffs(self):
+        raise NotImplementedError("get_coeffs not implemented for bdmce")
+
+    def tabulate(self, order, points, entity=None):
+
+        if entity is None:
+            entity = (self.ref_el.get_dimension(), 0)
+
+        entity_dim, entity_id = entity
+        transform = self.ref_el.get_entity_transform(entity_dim, entity_id)
+        points = list(map(transform, points))
+
+        phivals = {}
+
+        for o in range(order+1):
+            alphas = mis(2, o)
+            for alpha in alphas:
+                try:
+                    polynomials = self.basis[alpha]
+                except KeyError:
+                    polynomials = diff(self.basis[(0, 0)], *zip(variables, alpha))
+                    self.basis[alpha] = polynomials
+                T = np.zeros((len(polynomials[:, 0]), 2, len(points)))
+                for i in range(len(points)):
+                    subs = {v: points[i][k] for k, v in enumerate(variables[:2])}
+                    for j, f in enumerate(polynomials[:, 0]):
+                        T[j, 0, i] = f.evalf(subs=subs)
+                    for j, f in enumerate(polynomials[:, 1]):
+                        T[j, 1, i] = f.evalf(subs=subs)
```

Comment:
This should use `sympy.lambdify` so it's not horribly slow like the existing serendipity stuff.